src/gei/Services/GhesVersionCheckerService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Threading.Tasks;
+using OctoshiftCLI.Extensions;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Services
+{
+    public class GhesVersionCheckerService
+    {
+        private readonly OctoLogger _log;
+
+        public GhesVersionCheckerService(OctoLogger log)
+        {
+            _log = log;
+        }
+
+        public virtual async Task<bool> AreBlobCredentialsRequired(string ghesApiUrl, GithubApi api)
```

Comment:
`GithubApi` should definitely be injected into the service.

Reply:
If we do that we have to create a factory class too, which I wanted to avoid.

Reply:
It doesn't matter, because the `GithubApi` is a hard dependency and since we're doing DI we should inject it. I understand the pain of creating a factory but a good design surfaces the actual problems rather than suppressing them. Here our actual problem is the fact that we need a factory to create a `GithubApi` which can be addressed in a tech-debt issue later.

Reply:
as much as it pained me, factory class added